zebra-network/src/peer/handshake.rs

```diff
@@ -71,6 +73,15 @@ where
         self
     }
 
+    /// Provide a channel for registering inventory advertisements. Mandatory.
```

Comment:
Ideally this would be an optional parameter, handled the same way as the timestamp collector.